src/rgw/rgw_lc_s3.cc

```diff
@@ -32,7 +32,12 @@ bool LCExpiration_S3::xml_end(const char * el) {
   } else {
     date = lc_date->get_data();
     //We need return xml error according to S3
-    if (boost::none == ceph::from_iso_8601(date)) {
+    boost::optional<ceph::real_time> expiration_date = ceph::from_iso_8601(date);
```

Comment:
I'm ambivalent about this.  Just because Amazon limits the time to midnight doesn't mean we should, necessarily.  I'd prefer to truncate the time to midnight, if necessary, rather than throw an error.  I can be convinced otherwise, though.

Reply:
In my opinion, I'd prefer to keep the same with s3.